lib/ladybug/energy_model/schedule_ruleset_abridged.rb

```diff
@@ -0,0 +1,134 @@
+
+require 'ladybug/energy_model/model_object'
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class ScheduleRulesetAbridged < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash = {})
+        super(hash)
+
+        raise "Incorrect model type '#{@type}'" unless @type == 'ScheduleRulesetAbridged'
+      end
+
+      def defaults
+        result = {}
+        result
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getScheduleRulesetByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_schedule_ruleset = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+        openstudio_schedule_ruleset.setName(@hash[:name])
+        
+        if @hash[:summer_designday_schedule]
+          summer_designday_schedule = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+          values = @hash[:summer_designday_schedule][:values]
+          times = @hash[:summer_designday_schedule][:times]
+          values.each_index do |i|
+            summer_designday_schedule.addValue(OpenStudio::Time.new(0,times[i][0],times[i][1] ,0), values[i])
+          end 
+        end
+
+        openstudio_schedule_ruleset.setSummerDesignDaySchedule(summer_designday_schedule)
+
+        if @hash[:winter_designday_schedule]
+          winter_designday_schedule = OpenStudio::Model::ScheduleDay.new(openstudio_model)
+          values = @hash[:winter_designday_schedule][:values]
+          times = @hash[:winter_designday_schedule][:times]
+          values.each_index do |i|
+            winter_designday_schedule.addValue(OpenStudio::Time.new(0,times[i][0],times[i][1] ,0), values[i])
+          end 
+        end
+
+        openstudio_schedule_ruleset.setWinterDesignDaySchedule(winter_designday_schedule)
+        
+     
+        values_default_day = @hash[:default_day_schedule][:values]
+        times_default_day = @hash[:default_day_schedule][:times]
+
+        values_default_day.each_index do |i|
+          openstudio_schedule_ruleset.defaultDaySchedule.addValue(OpenStudio::Time.new(0,times_default_day[i][0],times_default_day[i][1] ,0), values_default_day[i])
+        end 
+
+        if @hash[:schedule_type_limit]
+          
+          schedule_type_limit_object = nil
+          schedule_type_limit = openstudio_model.getScheduleTypeLimitsByName(@hash[:schedule_type_limit])
+          unless schedule_type_limit.empty?
+            schedule_type_limit_object = schedule_type_limit.get
+            openstudio_schedule_ruleset.setScheduleTypeLimits(schedule_type_limit_object)
+          end
+        end
+
+        if @hash[:schedule_rules]
+          @hash[:schedule_rules].each do |rule|          
+            openstudio_schedule_rule = OpenStudio::Model::ScheduleRule.new(openstudio_schedule_ruleset)
```

Comment:
You might have already checked this but do you know for sure if this adds the rules in the right order?  In the Honeybee json, they are ordered from highest priority to lowest priority and I am not sure if this way of initializing ScheduleRules in the SDK preserves this order.

Reply:
@chriswmackey I've added code to assign the `ScheduleIndex` to the ScheduleRules. In OpenStudio, it follows the same prority 0 is the highest priority rule.